pkg/scheduler/schedule_one.go

```diff
@@ -99,9 +110,13 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	}
 
 	// bind the pod to its host asynchronously (we can do this b/c of the assumption step above).
+	//
+	// This goroutine becomes responsible for calling DonePod.
+	schedulingFailed = false
 	go func() {
 		bindingCycleCtx, cancel := context.WithCancel(ctx)
 		defer cancel()
+		defer sched.DonePod(assumedPodInfo.Pod.UID)
```

Comment:
`sched.DonePod` will be invoked twice?

Reply:
It should be only once. Because before proceeding to binding cycle, `schedulingFailed` is changed to false and L77 won't be executed.